paseos/visualization/space_animation.py

```diff
@@ -0,0 +1,280 @@
+                # spacecraft and ground stations behave differently and are plotted separately
+                if isinstance(obj.actor, SpacecraftActor):
+                    # update trajectory
+                    obj.plot.trajectory.set_data(data[-n_points:, :2].T)
+                    obj.plot.trajectory.set_3d_properties(data[-n_points:, 2].T)
+
+                    # update satellite position
+                    obj.plot.point.set_data_3d(data[-1, :])
+
+                    # update text box
+                    actor_info = self._populate_textbox(obj.actor)
+                    obj.plot.text.set_position_3d(data[-1, :] + self._textbox_offset)
+                    obj.plot.text.set_text(actor_info)
+
+                elif isinstance(obj.actor, GroundstationActor):
+                    # TODO: implement update of groundstation object
+                    pass
+            else:
+                if isinstance(obj.actor, SpacecraftActor):
+                    trajectory = self.ax.plot3D(data[0, 0], data[0, 1], data[0, 2])[0]
+                    obj.plot.trajectory = trajectory
+                    obj.plot.point = self.ax.plot(
+                        data[0, 0],
+                        data[0, 1],
+                        data[0, 2],
+                        "*",
+                        color=trajectory.get_color(),
+                    )[0]
+                    actor_info = self._populate_textbox(obj.actor)
+                    obj.plot.text = self.ax.text(
+                        data[0, 0] + self._textbox_offset,
+                        data[0, 1] + self._textbox_offset,
+                        data[0, 2] + self._textbox_offset,
+                        actor_info,
+                        bbox=dict(facecolor="white"),
+                        verticalalignment="bottom",
+                        clip_on=True,
+                    )
+
+                elif isinstance(obj.actor, GroundstationActor):
+                    # TODO: implement initial rendering of groundstation object
+                    pass
+
+        self.ax.set_box_aspect(
+            (
+                np.ptp(self.ax.get_xlim()),
+                np.ptp(self.ax.get_ylim()),
+                np.ptp(self.ax.get_zlim()),
+            )
+        )
+
+    def _make_actor_list(self, sim: PASEOS) -> List[BaseActor]:
+        """Arranges all actors (including the local) into a list
+        Args:
+        sim: simulation object.
+        Returns:
+            List[BaseActor]
+        """
+        if len(sim.known_actors) > 0:
+            current_actors = [sim.local_actor] + list(sim.known_actors.values())
+        else:
+            current_actors = [sim.local_actor]
+        return current_actors
+
+    def _update(self, sim: PASEOS) -> None:
+        """Updates the animation with the current actor information
+        Args:
+        sim (PASEOS): simulation object.
+        Returns:
+            None
+        """
+        logger.debug("Updating animation")
```

Comment:
How often is this called? potentially might be spamming debug a lot, maybe trace?

Reply:
Yes, updated.